### PR TITLE
generatorTemplate swagger

### PR DESCRIPTION
**Pull request by jgraff2**

### CODE REVIEW

File: api/v1/swagger.yaml [hunk 1]
```diff
@@ -1988,27 +1997,632 @@ paths:
...
+          $ref: "#/responses/400"
+        403:
+          $ref: "#/responses/403"
+        default:
+          $ref: "#/responses/genericError"
+
+  # ---------------------------------------------------------------------------
+
+  /generators/{key}:
+    x-swagger-router-controller: generators
```
Comment (shriramshankar) [inline api/v1/swagger.yaml]: are /generator/** swagger definitions from an older commit? The master branch has this already.

File: api/v1/swagger.yaml [hunk 2]
```diff
@@ -1988,27 +1997,632 @@ paths:
...
+                  Array of tag names. Tag name: Max 60 characters; case
+                  insensitive; allows alpha-numeric characters, underscore (_)
+                  and dash (-). Tag names cannot start with a dash (-).
+              connection:
+                type: object
+                description: Information about the connection.
+              context:
+                type: object
+                description: >-
+                  Variables accessible inside the transform function.
```
Comment (iamigo) [inline api/v1/swagger.yaml]: they're also available inside the toUrl fn *and* for variable substitution in connection.url

File: api/v1/swagger.yaml [hunk 3]
```diff
@@ -1988,27 +1997,632 @@ paths:
...
+              - -name
+              - -version
+              - -createdAt
+              - -updatedAt
+        -
+          $ref: "#/parameters/limitParam"
+        -
+          $ref: "#/parameters/offsetParam"
+        -
+          name: description
```
Comment (iamigo) [inline api/v1/swagger.yaml]: I don't think we are allowing query parameter filtering on description fields anywhere

File: api/v1/swagger.yaml [hunk 4]
```diff
@@ -1988,27 +1997,632 @@ paths:
...
+             a negative sign to indicate that a aspect with that tag should
+             be excluded. For example, ?tags=-BAZ,FOO will return only the
+             aspects with tag name not equal to BAZ or FOO. A mix of tag
+             names with and without minus signs, for example ?tags=BAZ,-FOO
+             are not allowed. GeneratorTemplates without tags are not included in the
+             include result set.
+          type: string
+          required: false
+        -
+          name: bulk
```
Comment (iamigo) [inline api/v1/swagger.yaml]: oops "bulk" needs to be moved _inside_ of the connection object as a connection attribute in the db model, and no need to have it here for query parameter filtering

File: api/v1/swagger.yaml [hunk 5]
```diff
@@ -1988,27 +1997,632 @@ paths:
...
+              name:
+                type: string
+                readOnly: true
+                description: >-
+                  The generatorTemplate's name.
+              version:
+                type: string
+                description: >-
+                  The version, in semantic versioning format ( MAJOR.MINOR.PATCH )
+              bulk:
```
Comment (iamigo) [inline api/v1/swagger.yaml]: see comment above about bulk

File: api/v1/swagger.yaml (hunk 1 above)
Comment (jgraff2) [inline api/v1/swagger.yaml]: I didn't add this. Looks like the diff is really messed up.

File: api/v1/swagger.yaml (hunk 2 above)
Comment (jgraff2) [inline api/v1/swagger.yaml]: will update.

File: api/v1/swagger.yaml (hunk 3 above)
Comment (jgraff2) [inline api/v1/swagger.yaml]: We are in aspects, subjects, and generators.
Should I remove them?

File: api/v1/swagger.yaml (hunk 4 above)
Comment (jgraff2) [inline api/v1/swagger.yaml]: ok, I’ll update the model in the other PR, and delete it here.

File: api/v1/swagger.yaml (hunk 2 above)
Comment (jgraff2) [inline api/v1/swagger.yaml]: This is actually on the generator. I didn't update this part, the diff is messed up. But the description is the same for contextDefinition on the generatorTemplate. I will update it there.

**Comment by iamigo** [comment]
@jgraff2 OK don't remove the filtering by description from the other resources, but let's not add it to any other resources

File: api/v1/swagger.yaml (hunk 3 above)
Comment (iamigo) [inline api/v1/swagger.yaml]: remove this one but not the others

**Pull request merged**